src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction
+  = (a, b) => String(a) > String(b)) {
+    const array = this;
+    const n = array.length;
+
+    for (let i = 0; i < n; i++) {
+      for (let j = 0; j < n - i - 1; j++) {
+        const compareResult = compareFunction
+          ? compareFunction(array[j], array[j + 1])
+          : String(array[j]).localeCompare(array[j + 1]);
```

Comment:
why do you need this check if you are adding a default value for compareFunction

Reply:
> why do you need this check if you are adding a default value for compareFunction
Thx , done